### PR TITLE
Draw should not interfere with dragPan.disable()

### LINKED ISSUE #654: Draw interferes dragPan.disable()
**mapbox-gl-js version**: 0.37
**mapbox-gl-draw version**: v0.17.4

### Steps to Trigger Behavior

 1. Load map.
 2. Load Draw control
 3. map.drapPan.disable()

### Expected Behavior
User can't drag pan the map

### Actual Behavior
Map is able to be dragpanned

If you comment out the 
            var draw = new MapboxDraw({
                displayControlsDefault: false,
                controls: {
                    polygon: true,
                    trash: true
                }
            });
You are able to disable dragPan as expected

### PR DESCRIPTION
**Pull request by MauPalantir**
Fixes #654

The issue is that Draw assumes that dragPan is enabled and simply enables it without question after certain operations (when dragpan is temporarily disabled). This change would set a variable called initialDragPanState and only re-enables when it was originally enabled.

### CODE REVIEW

File: src/modes/direct_select.js [hunk 1]
```diff
@@ -128,7 +132,8 @@ DirectSelect.onSetup = function(opts) {
     dragMoveLocation: opts.startPos || null,
     dragMoving: false,
     canDragMove: false,
-    selectedCoordPaths: opts.coordPath ? [opts.coordPath] : []
+    selectedCoordPaths: opts.coordPath ? [opts.coordPath] : [],
+    desiredDragPanState: null
```
Comment (stepankuzmin) [inline src/modes/direct_select.js:136]: Nit: probably this is some leftover variable?

File: src/setup.js [hunk 2]
```diff
@@ -64,10 +64,14 @@ export default function(ctx) {
 
       if (ctx.options.boxSelect) {
         map.boxZoom.disable();
+        const dragPanState = map.dragPan.isEnabled();
```
Comment (stepankuzmin) [inline src/setup.js:67]: Nit: 
```suggestion
        const dragPanIsEnabled = map.dragPan.isEnabled();
```

**Comment by stepankuzmin** [comment]
Hi @MauPalantir, thanks for the contribution! Overall this looks good to me, just a couple of nits.

**Comment by stepankuzmin** [comment]
Done in https://github.com/mapbox/mapbox-gl-draw/pull/1216

**Comment by lukeMichalakisTK** [comment]
@stepankuzmin has this fix been deployed? I am still experiencing this bug on mapbox-gl v3.7.0 and mapbox-gl-draw on v.1.4.3, which I believe are the most up to date versions?

**Comment by stepankuzmin** [comment]
Hey @lukeMichalakisTK, thanks for the bump.

This still needs to be released.

**Comment by lukeMichalakisTK** [comment]
@stepankuzmin No worries! We found a workaround locally that can hold us over in the meantime and I'll keep my eyes peeled for any releases from you guys, thanks!

**Pull request closed without merging**